ext/src/main/java/com/playone/mobile/common/NotValidVar.kt

```diff
@@ -0,0 +1,18 @@
+package com.playone.mobile.common
+
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class NotValidVar<T : Any>(private val invalidVal: T) : ReadWriteProperty<Any?, T> {
+    private var value: T = invalidVal
+
+    public override fun getValue(thisRef: Any?, property: KProperty<*>): T =
+        if (value != invalidVal)
+            value
+        else
+            throw IllegalStateException("Property ${property.name} should be initialized before get.")
```

Comment:
The msg is wrong. It should be `"Property ${property.name} value is invalid, please set a valid value."` or something blah blah blah.

Reply:
Also here shouldn't throw an exception for an invliad value I think.